### DISCUSSION TITLE
Alternative to `setup.py build_ext`

### DISCUSSION THREAD

**Discussion by codeandfire** [pypa/setuptools#3388] [Q&A] [answered]
Hi,

With commands of the form `python setup.py <command>` deprecated, is there any alternative to the `build_ext` command?

This question seems like something that might be covered/addressed somewhere, though I checked the docs and a number of other sources but couldn't find anything. Apologies in advance if I have missed something.

**Comment by abravalheri** [comment, marked as answer]
Hi @codeandfire, thank you very much for thoroughly exploring the docs and bringing attention to some points that might be problematic. It really helps.

I don't a replacement for "partial" commands is planned (i.e. commands that just build part of the distribution). Users can run `python -m build --wheel` to compile the entire wheel without passing through the sdist, but there is nothing currently planned to just build the extensions without the rest of the package.

I think that running `pip install -e .` might have as a side effect compiling the extensions "in-place" and leaving the binary files available under the working directory.

If there is a compelling use case we probably would need to discuss it more deeply (as far as I understand the general desire here would be that setuptools stop concerning about having to provide a CLI utility).

**Reply by codeandfire** [reply]
Right, that makes sense @abravalheri.

So does that imply that during development, the only way to recompile extension modules and put into effect any changes made to non-Python source code is to rerun `pip install -e .` ?

**Reply by abravalheri** [reply]
At least for the foreseeable future yes...

**Reply by codeandfire** [reply]
I see, thank you!
